pkg/filesystem/path/stringer.go

```diff
@@ -4,4 +4,5 @@ package path
 // converted to string representations.
 type Stringer interface {
 	GetUNIXString() string
+	GetWindowsString() string
```

Comment:
I have a feeling this should return `(string, error)`. Windows is a lot more restrictive than UNIX when it comes to representing paths. There are a whole bunch of reserved characters in filenames. There are also certain names that are forbidden (e.g., `LPT0`).

There may be cases in Buildbarn where we construct paths only using UNIX-style paths as inputs. But at the end we call `GetWindowsString()` to obtain a Windows-style path. That call should fail if the path is not representable.

Though I'm not requiring that you get this 100% watertight (that it rejects all paths that are not valid on Windows), the API should be prepared for it.